### PR TITLE
Add 2.14 release notes

### PR DESCRIPTION
**Pull request by jmazanec15**
### Description
Adds 2.14 release notes

### Check List
- [X] Commits are signed as per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/k-NN/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by navneet1v — APPROVED** [review]

**Comment by navneet1v** [comment]
@jmazanec15 can we ensure that CIs are successful before the merge

File: CHANGELOG.md
Comment (ryanbogan) [inline CHANGELOG.md:15]: Do we need to change this to 2.14?

**Review by naveentatikonda — APPROVED** [review]

File: CHANGELOG.md
Comment (jmazanec15) [reply CHANGELOG.md]: good catch

**Comment by jmazanec15** [comment]
> @jmazanec15 can we ensure that CIs are successful before the merge

@navneet1v In general, I agree they should finish, but there are a lot of known issues with CI right now. So I dont think we should hold up release notes PRs that do not change code while waiting for CI issues to be fixed.

**Comment by jmazanec15** [comment]
Mac is expected to fail with #1660

**Review by junqiu-lei — APPROVED** [review]
LGTM, thanks @jmazanec15

**Review by ryanbogan — APPROVED** [review]

**Pull request merged**
